Sources/Classes/Controllers/BaseCustomViewController.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  Copyright (c) 2018 Touch Instinct
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the Software), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED AS IS, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import UIKit.UIView
+
+/// Base controller configurable by view model and custom view.
+open class BaseCustomViewController<ViewModel, View: UIView>: BaseConfigurableController<ViewModel> {
+
+    /// Contained custom view.
+    public let customView: View
+
+    /// Initializer with view model and custom view parameters.
+    ///
+    /// - Parameters:
+    ///   - viewModel: A view model to configure this controller.
+    ///   - customView: UIView instance to assign in view property.
+    public init(viewModel: ViewModel, customView: View) {
```

Comment:
Are you sure? Creation view before it needs.

Reply:
what kind of problems can be?